test/integration/exec_int_test.go

```diff
@@ -165,6 +166,28 @@ echo "Hello $name!"
 	cp.ExpectExitCode(0)
 }
 
+func (suite *ExecIntegrationTestSuite) TestExec_SpaceInCacheDir() {
+	suite.OnlyRunForTags(tagsuite.Exec)
+	ts := e2e.New(suite.T(), false)
+	defer ts.Close()
+
+	cacheDir := filepath.Join(ts.Dirs.Cache, "dir with spaces")
+	err := fileutils.MkdirUnlessExists(cacheDir)
+	suite.Require().NoError(err)
+
+	cp := ts.SpawnWithOpts(
+		e2e.AppendEnv(fmt.Sprintf("%s=%s", constants.CacheEnvVarName, cacheDir)),
+		e2e.AppendEnv(fmt.Sprintf(`%s=""`, constants.DisableRuntime)),
+		e2e.WithArgs("activate", "ActiveState-CLI/Python3"),
+	)
+
+	cp.SendLine("python3 --version")
+	cp.Expect("Python 3.")
+
+	cp.SendLine("exit")
+	cp.ExpectExitCode(0)
+}
```

Comment:
This feels like it belongs in activate_int_test.

Reply:
I thought about that, but the focus is particularly on the effectiveness of the executor .sh and .bat files that are used within an activated state.

Reply:
`exec_int_test` focusses on `state exec`. Not to be confused with executors that use `state exec` under the hood. The latter is already covered in activate_int_test.